sklearn/impute/tests/test_impute.py

```diff
@@ -1383,3 +1383,67 @@ def test_imputation_order(order, idx_order):
                                random_state=0).fit(X)
         idx = [x.feat_idx for x in trs.imputation_sequence_]
         assert idx == idx_order
+
+
+@pytest.mark.parametrize("missing_value", [-1, np.nan])
+def test_simple_imputation_inverse_transform(missing_value):
+    # Test inverse_transform feature for np.nan
+    X_1 = np.array([
+        [9, missing_value, 3, -1],
+        [4, -1, 5, 4],
+        [6, 7, missing_value, -1],
+        [8, 9, 0, missing_value]
+    ])
+
+    X_2 = np.array([
+        [5, 4, 2, 1],
+        [2, 1, missing_value, 3],
+        [9, missing_value, 7, 1],
+        [6, 4, 2, missing_value]
+    ])
+
+    X_3 = np.array([
+        [1, missing_value, 5, 9],
+        [missing_value, 4, missing_value, missing_value],
+        [2, missing_value, 7, missing_value],
+        [missing_value, 3, missing_value, 8]
+    ])
+
+    X_4 = np.array([
+        [1, 1, 1, 3],
+        [missing_value, 2, missing_value, 1],
+        [2, 3, 3, 4],
+        [missing_value, 4, missing_value, 2]
+    ])
+
+    imputer = SimpleImputer(missing_values=missing_value, strategy='mean',
+                            add_indicator=True)
+
+    X_1_trans = imputer.fit_transform(X_1)
+    X_1_orig = imputer.inverse_transform(X_1_trans)
+
+    X_2_trans = imputer.transform(X_2)
+    X_2_orig = imputer.inverse_transform(X_2_trans)
+
+    assert_array_equal(X_1_orig, X_1)
+    assert_array_equal(X_2_orig, X_2)
+
+    for X in [X_3, X_4]:
+        X_trans = imputer.fit_transform(X)
+        X_orig = imputer.inverse_transform(X_trans)
+        assert_array_equal(X_orig, X)
+
+
+@pytest.mark.parametrize("missing_value", [-1, np.nan])
+def test_simple_imputation_inverse_transform_exceptions(missing_value):
+    X_1 = np.array([
+        [9, missing_value, 3, -1],
+        [4, -1, 5, 4],
+        [6, 7, missing_value, -1],
+        [8, 9, 0, missing_value]
+    ])
+    with pytest.raises(ValueError, match="add_indicator=True"):
+        imputer = SimpleImputer(missing_values=missing_value,
+                                strategy="mean")
+        X_1_trans = imputer.fit_transform(X_1)
```

Comment:
move these line outside from the raises. It will be more explicit to know which line is raising the error